src/query/interpret/eval.hpp

```diff
@@ -903,7 +903,15 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
     return TypedValue(std::move(result), ctx_->memory);
   }
 
-  TypedValue Visit(Exists &exists) override { return TypedValue{frame_->at(symbol_table_->at(exists)), ctx_->memory}; }
+  TypedValue Visit(Exists &exists) override {
+    TypedValue &frame_exists_value = frame_->at(symbol_table_->at(exists));
+    if (frame_exists_value.IsFunction()) [[likely]] {
+      TypedValue result{ctx_->memory};
+      frame_exists_value.ValueFunction()(&result);
+      return result;
+    }
+    return TypedValue{frame_exists_value, ctx_->memory};
```

Comment:
I would rather throw here, since this should be a function at any point in time at the moment.

Reply:
or `MG_ASSERT`